cmake/OpenCVConfig.cmake

```diff
@@ -47,7 +47,7 @@ endif()
 
 if(NOT DEFINED OpenCV_STATIC)
   # look for global setting
-  if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
+  if(BUILD_SHARED_LIBS)
```

Comment:
This change only affects the behaviour in case OpenCV was not built with ```BUILD_SHARED_LIBS=ON``` and the user did not explicitly define ```OpenCV_STATIC``` in his application's build prior to using ```find_package( OpenCV )```. So what you want to preserve is the behaviour of the package not being found in this case instead of finding the static libs, correct? That is a little strange (since I consider this simply a bug), but I will not deny that existing builds might rely on the fail.

I will remove the change from this PR. Thanks for your feedback.